### ISSUE TITLE
Static content deploy - Don't shows error message, just stack trace

### ISSUE THREAD

**Issue by ihor-sviziev** [magento/magento2#22882]
### Preconditions (*)
1. Magento 2.2.7
2. Configured Build system according to [technical details](https://devdocs.magento.com/guides/v2.2/config-guide/deployment/pipeline/technical-details.html).

### Steps to reproduce (*)
1. Emulate exception that happened in https://github.com/magento/magento2/issues/22880 by adding above this line 
https://github.com/magento/magento2/blob/15e3c07f3aeb94d4147e49401d409afc3dcd93c9/lib/internal/Magento/Framework/View/Asset/LockerProcess.php#L89
throwing exception:
```
throw new FileSystemException(__('Test'));
```
2. Run `php bin/magento setup:static-content:deploy -f --theme Magento/blank`

### Expected result (*)
1. In case of any errors - there should see exception details, not only stacktrace

### Actual result (*)
1. In case of error we can see stack trace, but not exception message
![Capture 2019-05-14 at 12 07 48](https://user-images.githubusercontent.com/1873745/57685577-de3c7780-7640-11e9-9503-0dc4f6f843db.png)

```
#0 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/framework/Filesystem/Driver/File.php(399): FilesystemIterator->__construct('/opt/atlassian/...', 12288)
#1 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/framework/Filesystem/Directory/Write.php(188): Magento\Framework\Filesystem\Driver\File->deleteDirectory('/opt/atlassian/...')
#2 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/framework/View/Asset/LockerProcess.php(89): Magento\Framework\Filesystem\Directory\Write->delete('tmp/alternative...')
#3 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/framework/View/Asset/PreProcessor/AlternativeSource.php(116): Magento\Framework\View\Asset\LockerProcess->unlockProcess()
#4 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/module-developer/Model/View/Asset/PreProcessor/PreprocessorStrategy.php(76): Magento\Framework\View\Asset\PreProcessor\AlternativeSource->process(Object(Magento\Framework\View\Asset\PreProcessor\Chain))
#5 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/framework/View/Asset/PreProcessor/Pool.php(77): Magento\Developer\Model\View\Asset\PreProcessor\PreprocessorStrategy->process(Object(Magento\Framework\View\Asset\PreProcessor\Chain))
#6 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/framework/View/Asset/Source.php(152): Magento\Framework\View\Asset\PreProcessor\Pool->process(Object(Magento\Framework\View\Asset\PreProcessor\Chain))
#7 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/framework/View/Asset/Source.php(105): Magento\Framework\View\Asset\Source->preProcess(Object(Magento\Framework\View\Asset\File))
#8 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/framework/View/Asset/File.php(157): Magento\Framework\View\Asset\Source->getFile(Object(Magento\Framework\View\Asset\File))
#9 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/framework/App/View/Asset/Publisher.php(73): Magento\Framework\View\Asset\File->getSourceFile()
#10 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/framework/App/View/Asset/Publisher.php(61): Magento\Framework\App\View\Asset\Publisher->publishAsset(Object(Magento\Framework\View\Asset\File))
#11 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/module-deploy/Service/DeployStaticFile.php(92): Magento\Framework\App\View\Asset\Publisher->publish(Object(Magento\Framework\View\Asset\File))
#12 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/module-deploy/Service/DeployPackage.php(189): Magento\Deploy\Service\DeployStaticFile->deployFile('css/email.less', Array)
#13 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/module-deploy/Service/DeployPackage.php(136): Magento\Deploy\Service\DeployPackage->processFile(Object(Magento\Deploy\Package\PackageFile), Object(Magento\Deploy\Package\Package))
#14 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/module-deploy/Service/DeployPackage.php(107): Magento\Deploy\Service\DeployPackage->deployEmulated(Object(Magento\Deploy\Package\Package), Array, true)
#15 [internal function]: Magento\Deploy\Service\DeployPackage->Magento\Deploy\Service\{closure}()
#16 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/framework/App/State.php(186): call_user_func_array(Object(Closure), Array)
#17 /opt/atlassian/pipelines/agent/build/tmp/magento2/generated/code/Magento/Framework/App/State/Interceptor.php(89): Magento\Framework\App\State->emulateAreaCode('frontend', Object(Closure), Array)
#18 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/module-deploy/Service/DeployPackage.php(108): Magento\Framework\App\State\Interceptor->emulateAreaCode('frontend', Object(Closure))
#19 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/module-deploy/Process/Queue.php(319): Magento\Deploy\Service\DeployPackage->deploy(Object(Magento\Deploy\Package\Package), Array, true)
#20 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/module-deploy/Process/Queue.php(235): Magento\Deploy\Process\Queue->execute(Object(Magento\Deploy\Package\Package))
#21 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/module-deploy/Process/Queue.php(216): Magento\Deploy\Process\Queue->executePackage(Object(Magento\Deploy\Package\Package), 'frontend/My/Th...', Array, false)
#22 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/module-deploy/Process/Queue.php(169): Magento\Deploy\Process\Queue->assertAndExecute('frontend/My/Th...', Array, Array)
#23 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/module-deploy/Strategy/QuickDeploy.php(76): Magento\Deploy\Process\Queue->process()
#24 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/module-deploy/Service/DeployStaticContent.php(109): Magento\Deploy\Strategy\QuickDeploy->deploy(Array)
#25 /opt/atlassian/pipelines/agent/build/tmp/magento2/setup/src/Magento/Setup/Console/Command/DeployStaticContentCommand.php(140): Magento\Deploy\Service\DeployStaticContent->deploy(Array)
#26 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/symfony/console/Command/Command.php(245): Magento\Setup\Console\Command\DeployStaticContentCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#27 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/symfony/console/Application.php(835): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#28 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/symfony/console/Application.php(185): Symfony\Component\Console\Application->doRunCommand(Object(Magento\Setup\Console\Command\DeployStaticContentCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#29 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/magento/framework/Console/Cli.php(102): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#30 /opt/atlassian/pipelines/agent/build/tmp/magento2/vendor/symfony/console/Application.php(117): Magento\Framework\Console\Cli->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#31 /opt/atlassian/pipelines/agent/build/tmp/magento2/bin/magento(23): Symfony\Component\Console\Application->run()
#32 {main}
```

This issue was discovered during testing https://github.com/magento/magento2/pull/22607
and working on reporting https://github.com/magento/magento2/issues/22880

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev. Thank you for your report. 
The issue has been fixed in magento/magento2#22884 by @ihor-sviziev in 2.3-develop branch
Related commit(s):
 - [881e92728c3b96b9afdcbc4347bc2d75a7b31778](https://github.com/magento/magento2/commit/881e92728c3b96b9afdcbc4347bc2d75a7b31778)

The fix will be available with the upcoming 2.3.3 release.

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev. Thank you for your report. 
The issue has been fixed in magento/magento2#23114 by @ihor-sviziev in 2.2-develop branch
Related commit(s):
 - [38b8333fad0506816da58276df73d6dee36a07b2](https://github.com/magento/magento2/commit/38b8333fad0506816da58276df73d6dee36a07b2)

The fix will be available with the upcoming 2.2.10 release.
